source/Subsystem/Renderer/World.cpp

```diff
@@ -24,12 +25,23 @@ void World::LoadDummyData()
 	Model* model = myAssetLoader->LoadModel("../Data/Models/Planet/Planet.obj");
 	if (!model)
 	{
-		printf("Error loading dummy data");
+		Log::Print(LogType::PROBLEM, "Error loading dummy data");
 		return;
 	}
 
-	model->myTextureUnit = 0;
-	myTextureLibrary->CompileTexture(model->myTextures[0]);
+	for (const std::pair<std::string, TextureType> texturePair : model->myTextureMap)
```

Comment:
make it a const pair ref so that you don't need to copy the string and texture type on each iteration.